### PR TITLE
fix some filter parsing issues

### PR DESCRIPTION
**Pull request by hjanetzek**

### CODE REVIEW

**Comment by matteblair** [comment]
https://mapzen.com/documentation/tangram/Filters-Overview/#lists-imply-any-mappings-imply-all

If we were to make these filters more explicit, we should add `any`, not `all`. However, these seem to already be well-formed filters according to our docs, so we should fix this in tangram-es rather than here.

File: layers/places.yaml [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ layers:
                 any:
                     - source: [openstreetmap, openstreetmap.org]
                     - source: [whosonfirst, whosonfirst.mapzen.com]
```
Comment (bcamper) [inline layers/places.yaml]: This won't work with the `source` key listed multiple times.

File: layers/roads.yaml [hunk 2]
```diff
@@ -321,9 +323,10 @@ layers:
 
         minor_road:
             filter:
-                - kind: minor_road
-                - not: { kind_detail: service }
-                - $zoom: { min: 8 }
+                all:
```
Comment (bcamper) [inline layers/roads.yaml]: This doesn't need to be nested in `all` either, can just be this -- **assuming the intent is to only apply the layer when all of the conditions are true**:

```
             filter:
                kind: minor_road
                not: { kind_detail: service }
                $zoom: { min: 8 }
```

File: layers/roads.yaml [hunk 3]
```diff
@@ -164,8 +165,9 @@ layers:
 
         major_road:
             filter:
-                - kind: major_road
-                - $zoom: { min: 8 }
+                all:
```
Comment (bcamper) [inline layers/roads.yaml]: Doesn't need nested `all` (see comment below).

File: layers/roads.yaml [hunk 4]
```diff
@@ -56,8 +56,9 @@ layers:
 
         highway:
             filter:
-                - kind: highway
-                - $zoom: { min: 8 }
+                all:
```
Comment (bcamper) [inline layers/roads.yaml]: Doesn't need nested `all` (see comment below).

**Comment by matteblair** [comment]
Looking at these filters in context now it seems likely that these were not intended to be used as 'any' filters, they were probably put in a sequence by accident. So now this suggested change makes sense to me.

File: layers/places.yaml (hunk 1 above)
Comment (nvkelso) [reply layers/places.yaml]: The `openstreetmap` source line can be removed completely.

**Comment by nvkelso** [comment]
The various comments here confuse me... I think the final result is we **do** want to merge this PR to improve the TRON style?

File: layers/roads.yaml [hunk 5]
```diff
@@ -56,8 +56,8 @@ layers:
 
         highway:
             filter:
-                - kind: highway
-                - $zoom: { min: 8 }
+                kind: highway
+                $zoom: { min: 8 }
             highway-z8:
```
Comment (bcamper) [inline layers/roads.yaml:61]: @nvkelso is the intent here that the filter is true when `kind: highway` **AND** zoom >= 8 (what the PR changes to), or when either `kind: highway` **OR** zoom >= 8 (what it was before)?

**Comment by bcamper** [comment]
@nvkelso what is the intent of this filter?

```
        neighborhood-z13:
            filter:
                all:
                    - name: true
                    - kind: [macrohood, neighbourhood]
                    - $zoom: [13]
                any:
                    - source: [whosonfirst, whosonfirst.mapzen.com]
                    - all:
                        - min_zoom: [11,12,13]
                        - max_zoom: { min: 14 }
                        - is_landuse_aoi: false
                        - kind_tile_rank: { max: 8 }
```

File: layers/roads.yaml (hunk 5 above)
Comment (nvkelso) [reply layers/roads.yaml:61]: AND

**Comment by nvkelso** [comment]
@bcamper Regarding the `neighborhood-z13` block, it's left over from when we also supported OpenStreetMap neighbourhoods. 

I think this is the intent:

```
neighborhood-z13:
            filter:
                all:
                    - name: true
                    - kind: [macrohood, neighbourhood]
                    - $zoom: [13]
                    - min_zoom: [11,12,13]
                    - max_zoom: { min: 14 }
                    - is_landuse_aoi: false
                    - kind_tile_rank: { max: 8 }
```

**Comment by bcamper** [comment]
Got it, then this can be more succinctly written without the `all`, which is implied by default for key/value mappings:

```
neighborhood-z13:
            filter:
                 name: true
                 kind: [macrohood, neighbourhood]
                 $zoom: [13]
                 min_zoom: [11,12,13]
                 max_zoom: { min: 14 }
                 is_landuse_aoi: false
                 kind_tile_rank: { max: 8 }
```

**Comment by nvkelso** [comment]
Simplified neighbourhood `all` filters in 71ab3d7f6b334359d03eb02092c02b02bc2be9d0.

File: layers/roads.yaml (hunk 5 above)
Comment (nvkelso) [reply layers/roads.yaml:61]: LGTM

**Comment by nvkelso** [comment]
LGTM

**Pull request merged**